numpy/__init__.py

```diff
@@ -155,70 +155,32 @@
     from . import matrixlib as _mat
     from .matrixlib import *
 
-    # Deprecations introduced in NumPy 1.20.0, 2020-06-06
-    import builtins as _builtins
-
+    # Future warning introduced in NumPy 1.24.0, 2022-11-17
     _msg = (
-        "`np.{n}` is a deprecated alias for the builtin `{n}`. "
-        "To silence this warning, use `{n}` by itself. Doing this will not "
-        "modify any behavior and is safe. {extended_msg}\n"
-        "Deprecated in NumPy 1.20; for more details and guidance: "
-        "https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations")
-
-    _specific_msg = (
-        "If you specifically wanted the numpy scalar type, use `np.{}` here.")
-
-    _int_extended_msg = (
-        "When replacing `np.{}`, you may wish to use e.g. `np.int64` "
-        "or `np.int32` to specify the precision. If you wish to review "
-        "your current use, check the release note link for "
-        "additional information.")
+        "`np.{n}` is a deprecated alias for `{an}`.  (Deprecated NumPy 1.24)")
 
+    # Some of these are awkard (since `np.str` may be preferable in the long
+    # term), but overall the names ending in 0 seem undesireable
     _type_info = [
-        ("object", ""),  # The NumPy scalar only exists by name.
-        ("bool", _specific_msg.format("bool_")),
-        ("float", _specific_msg.format("float64")),
-        ("complex", _specific_msg.format("complex128")),
-        ("str", _specific_msg.format("str_")),
-        ("int", _int_extended_msg.format("int"))]
+        ("bool8", bool_, "np.bool_"),
+        ("int0", intp, "np.intp"),
+        ("uint0", uintp, "np.uintp"),
+        ("str0", str_, "np.str_"),
+        ("bytes0", bytes_, "np.bytes_"),
+        ("void0", void, "np.void"),
+        ("object0", object_,
+            "`np.object0` is a deprecated alias for `np.object_`. "
+            "`object` can be used instead.  (Deprecated NumPy 1.24)")]
+
+    # Some of these could be defined right away, but most were aliases to
+    # the Python objects before.  When defined, these should possibly not
+    # be added to `__all__` to avoid import with `from numpy import *`.
```

Comment:
May be useful to add a more explicit warning / timescale now? E.g., "Some of these were removed in 1.24.0, don't redefine them before 1.27.0 / 2.0".